keps/sig-node/4639-oci-volume-source/README.md

```diff
@@ -858,6 +869,27 @@ in back-to-back releases.
 - Deprecate the flag
 -->
 
+#### Alpha
+
+- Initial implementation added
+- CRI implementations add support
+
+#### Beta
+
+- Add support for [`subPath`](https://kubernetes.io/docs/concepts/storage/volumes/#using-subpath)
```

Comment:
For other volume types, the subpath for a container mount is maintained by the kubelet. kubelet generate the computed hostPath if a subpath is provided and pass it into CRI, so the CRI doesn't know the subpath. But the image volume is different. If we want to support subpath for image volume, we need to pass the subpath to CRI runtime.  Then the CRI runtime should handle the subpath for image volume. 

BTW, the validation on pod spec disallows the subpath for image volume now. We need to remove this limitation and add more detailed design for subpath support for image volume in this KEP.

Reply:
How about reusing the `hostPath` variable of the mount for the sub path?
https://github.com/kubernetes/cri-api/blob/010fdf8d785a7f35fa5a1cc80c40bde0c13e1dfd/pkg/apis/runtime/v1/api.proto#L224-L227

It's empty if `image != nil` per: https://github.com/kubernetes/kubernetes/blob/0b91def6bcbdde82206ec779791762fb6bc337af/pkg/kubelet/kubelet_pods.go#L267-L274

Reply:
```
// Mount specifies a host volume to mount into a container.
message Mount {
    // Path of the mount within the container.
    string container_path = 1;
    // Path of the mount on the host. If the hostPath doesn't exist, then runtimes
    // should report error. If the hostpath is a symbolic link, runtimes should
    // follow the symlink and mount the real destination to container.
    string host_path = 2;
```

According to cri spec, we can not reuse the host_path for image volume type. kubelet doesn't know the actual path which is determined by the runtime.

Reply:
>  The idea was to use the host_path field (empty for image volumes right now) to carry the sub path value. 

It should work but the variable `host_path` would be misleading for developers. I would suggest using a new field instead, i.e. `image_sub_path` or `image_path`. 

And I'm happy to hear more ideas and suggestions from you and others.

Reply:
The image is mounted with read-only mode. If the subpath is provided but it doesn't exist in the image, what will be happened?

Reply:
I also am a fan of reusing but am also happy to introduce a new field. the key is if subpath isn't used, older cri implemetations should still work IMO. I think if subpath doesn't exist in image the pod creation should fail

Reply:
I'll update the section to cover the sub path details.